File: Kotlin/프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.kt
class Solution {
    fun solution(lottos: IntArray, win_nums: IntArray): IntArray {
        var answer: IntArray = IntArray(2)

        var maxCount = 0
        for (lotto in lottos) {
            for (winNum in win_nums) {
                if (lotto == winNum) {
                    answer[1]++
                    break
                }
                if (lotto == 0) {
                    maxCount++
                    break
                }
            }
        }


        answer[0] =  answer[1] + maxCount

        for (i in answer.indices) {
            when (answer[i]) {
                6 -> answer[i] = 1
                5 -> answer[i] = 2
                4 -> answer[i] = 3
                3 -> answer[i] = 4
                2 -> answer[i] = 5
                else -> answer[i] = 6
            }
        }

        return answer
    }
}

File: C#/프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.cs
using System;

public class Solution {
    public int[] solution(int[] lottos, int[] win_nums) {
        int[] answer = new int[2];
        
        Array.Sort(lottos, (a,b) => b.CompareTo(a));
        Array.Sort(win_nums, (a,b) => b.CompareTo(a));
        
        int zero = 0;
        int corr = 0;
        
        for(int i = 0; i < 6; i++)
        {
            if(lottos[i] == 0){
                zero++;
                continue;
            }
            
            if(Array.IndexOf(win_nums, lottos[i]) > -1)
            {
                    corr++;
            }
        }
        
        int maxCor = corr + zero;
        
        answer[0] = 6 - maxCor == 6 ? 6 : 6 - maxCor + 1;
        answer[1] = 6 - corr == 6 ? 6 : 6 - corr + 1;
        
        return answer;
    }
}